app/components/link_disabled_by_tooltip_component.html.erb

```diff
@@ -0,0 +1,3 @@
+<%= link_to_if enabled?, text, path, html_options do %>
+  <%= content_tag(:span, text, html_options) %>
```

Comment:
are the link tag and content tag's html options always the same?

Reply:
Kinda yes but no. It decides whether to make a link or a span depending on the value of `enabled?`. The keys and values inside `html_options` will also be different depending on the value of `enabled?`. 

If enabled is false (when we render a span) then there will be items inside `html_options` for the tooltip behavior, and the class list will contain a `disabled` class. If enabled is true, there will be no tooltip behavior items, but there will be link-specific stuff in there (such as `method`)

Reply:
I suppose another way to do this would be two different methods... `link_html_options` and `span_html_options`?

Reply:
@awead I refactored `html_options` into two different methods (one for enabled, and one for disabled) and the code is MUCH cleaner!